packages/toolkit/src/tsHelpers.ts

```diff
@@ -205,3 +205,7 @@ export type Id<T> = { [K in keyof T]: T[K] } & {}
 export type Tail<T extends any[]> = T extends [any, ...infer Tail]
   ? Tail
   : never
+
+export type KeysMatching<T, V> = {
```

Comment:
```suggestion
export type KeysForValueOfType<T, V> = {
```
`KeysMatching` sounded to me like `keyof T & keyof V`